modules/bash-commons/src/aws.sh

```diff
@@ -20,36 +20,65 @@ readonly three_hours_in_s=10800
 # a token from IMDSv2
 readonly six_hours_in_s=21600
 
-# Detect if the instance is using IMDSv2 or if it is using IMDSv1 still.
+# shellcheck source=./modules/bash-commons/src/assert.sh
+source "$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)/assert.sh"
+# shellcheck source=./modules/bash-commons/src/log.sh
+source "$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)/log.sh"
+
+# Set variable to an empty string if it is unbound to prevent "unbound variable".
+export GRUNTWORK_BASH_COMMONS_IMDS_VERSION=${GRUNTWORK_BASH_COMMONS_IMDS_VERSION-}
+# Set default variable to version "2" if nothing is set.
+export default_instance_metadata_version=${default_instance_metadata_version-"2"}
```

Comment:
If nothing is set for the default IMDS version, set it to 2.  The calling script can override this by setting either of the two variables:
- GRUNTWORK_BASH_COMMONS_IMDS_VERSION
- default_instance_metadata_version